library/kani/src/exe_trace.rs

```diff
@@ -24,6 +24,16 @@ pub fn exe_trace_run<F: Fn()>(mut local_det_vals: Vec<Vec<u8>>, proof_harness: F
     });
     // Since F is a type argument, there should be a direct, static call to proof_harness().
     proof_harness();
+    // This code will not run if a user assertion fails on deterministic playback.
+    // But if you comment out the failing assertion during playback,
+    // this can be used to know if too many det vals were loaded into the deterministic test case.
+    DET_VALS.with(|glob_det_vals| {
+        let ref_glob_det_vals = &*glob_det_vals.borrow();
+        assert!(
+            ref_glob_det_vals.is_empty(),
+            "At the end of deterministic playback, there were still these deterministic values left over `{:?}`", ref_glob_det_vals
```

Comment:
Maybe say that this can happen because:
1. The code being verified has changed and the user should update their generated tests.
2. There is a bug on kani. Please report the issue here: https://github.com/model-checking/kani/issues/new?assignees=&labels=bug&template=bug_report.md

Once you create the documentation, you could replace this by a "for more details, see `<link-to-doc>`".

Reply:
Good suggestion. Done